src/registrar/migrations/0121_alter_domaininformation_submitter_and_more.py

```diff
@@ -0,0 +1,38 @@
+# Generated by Django 4.2.10 on 2024-08-29 23:17
+
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("registrar", "0120_add_domainrequest_submission_dates"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="domaininformation",
+            name="submitter",
+            field=models.ForeignKey(
+                blank=True,
+                help_text='Person listed under "your contact information" in the request form',
+                null=True,
+                on_delete=django.db.models.deletion.CASCADE,
+                related_name="submitted_domain_requests_information",
+                to="registrar.contact",
+            ),
+        ),
+        migrations.AlterField(
```

Comment:
Bug: this migration isn't being recognized so I can't run the following migrations because Django thinks DomainRequest.submitter is still a protected field - leaving space for suggestions while I'm out tomorrow

Reply:
@erinysong what you have here would deleted all domain information objects if their submitter is deleted and, below, it will delete all domain requests when their submitter is deleted. This would be a critical error on stable that would require a database restore.
This makes sense why this wasn't working as it was trying to delete requests but it cannot blocks this as other items would need to be deleted first.


Going the code route you have below where you are manually deleting means you do not have to try to mess with cascade. Using cascade would require a different migration setup